internal/server/config.go

```diff
@@ -686,13 +686,18 @@ func (s Server) loadProviders(db data.GormTxn, providers []Provider) error {
 	return nil
 }
 
-func (Server) loadProvider(db data.GormTxn, input Provider) (*models.Provider, error) {
+func (s Server) loadProvider(db data.GormTxn, input Provider) (*models.Provider, error) {
 	// provider kind is an optional field
 	kind, err := models.ParseProviderKind(input.Kind)
 	if err != nil {
 		return nil, fmt.Errorf("could not parse provider in config load: %w", err)
 	}
 
+	clientSecret, err := secrets.GetSecret(input.ClientSecret, s.secrets)
```

Comment:
Where did we do this lookup previously? I think this could be a breaking change for anyone on-prem that is using a  `:` in their client secret #3333

Reply:
This wouldn't have worked before either so it wont be a breaking change in this case